src/enhancers/enhance_co2.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{
+    model::{self, Collections},
+    objects::PhysicalMode,
+};
+use typed_index_collection::CollectionWithId;
+
+lazy_static::lazy_static! {
+    static ref CO2_EMISSIONS: std::collections::HashMap<&'static str, f32> = {
+        let mut modes_map = std::collections::HashMap::new();
+        modes_map.insert(model::AIR_PHYSICAL_MODE, 144.6f32);
+        modes_map.insert(model::BIKE_PHYSICAL_MODE, 0f32);
+        modes_map.insert(model::BIKE_SHARING_SERVICE_PHYSICAL_MODE, 0f32);
+        // Unknown value
+        // modes_map.insert(model::"Boat", 0.0f32);
+        modes_map.insert(model::BUS_PHYSICAL_MODE, 132f32);
+        modes_map.insert(model::BUS_RAPID_TRANSIT_PHYSICAL_MODE, 84f32);
+        modes_map.insert(model::CAR_PHYSICAL_MODE, 184f32);
+        modes_map.insert(model::COACH_PHYSICAL_MODE, 171f32);
+        modes_map.insert(model::FERRY_PHYSICAL_MODE, 279f32);
+        modes_map.insert(model::FUNICULAR_PHYSICAL_MODE, 3f32);
+        modes_map.insert(model::LOCAL_TRAIN_PHYSICAL_MODE, 30.7f32);
+        modes_map.insert(model::LONG_DISTANCE_TRAIN_PHYSICAL_MODE, 3.4f32);
+        modes_map.insert(model::METRO_PHYSICAL_MODE, 3f32);
+        modes_map.insert(model::RAPID_TRANSIT_PHYSICAL_MODE, 6.2f32);
+        // Unknown value
+        // modes_map.insert(model::RailShuttle, 0.0f32);
+        // Unknown value
+        // modes_map.insert(model::Shuttle, 0.0f32);
+        // Unknown value
+        // modes_map.insert(model::SuspendedCableCar, 0.0f32);
+        modes_map.insert(model::TAXI_PHYSICAL_MODE, 184f32);
+        modes_map.insert(model::TRAIN_PHYSICAL_MODE, 11.9f32);
+        modes_map.insert(model::TRAMWAY_PHYSICAL_MODE, 4f32);
+        modes_map
+    };
+}
+
+/// Physical mode should contains CO2 emissions. If the values are not present
+/// in the NTFS, some default values will be used.
+pub fn enhance_with_co2(collections: &mut Collections) {
```

Comment:
I'm never sure which is better between

```rust
pub fn enhance_with_co2(collections: &mut Collections) { ... }
```

and 

```rust
pub fn enhance_with_co2(collections: Collections) -> Collections { ... }
```

The benefit of the second version is that the function is pure.

Reply:
sure but I wanted to have minimal changes and I'm not really convinced one is really better than the other :thinking:

Reply:
And you were probably right to do minimal changes at this point. I guess I took the opportunity to review it since the code was changed. Anyone else has an opinion on the topic?